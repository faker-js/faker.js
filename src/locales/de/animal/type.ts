export default [
  'Adler',
  'Affe',
  'Biene',
  'Bär',
  'Delfin',
  'Eichhörnchen',
  'Eisbär',
  'Elefant',
  'Fisch',
  'Flamingo',
  'Fledermaus',
  'Frosch',
  'Fuchs',
  'Gecko',
  'Giraffe',
  'Gorilla',
  'Hai',
  'Hamster',
  'Hund',
  'Kaninchen',
  'Katze',
  'Koala',
  'Krokodil',
  'Kuh',
  'Känguru',
  'Löwe',
  'Nashorn',
  'Nilpferd',
  'Panda',
  'Papagei',
  'Pfau',
  'Pferd',
  'Pinguin',
  'Reh',
  'Schildkröte',
  'Schlange',
  'Schmetterling',
  'Seelöwe',
  'Strauß',
  'Tiger',
  'Vogel',
  'Wal',
  'Wolf',
  'Zebra',
];
