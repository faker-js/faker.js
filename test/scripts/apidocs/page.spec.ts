import { describe, expect, it } from 'vitest';
import { toRefreshFunction } from '../../../scripts/apidocs/output/page';
import type { RawApiDocsMethod } from '../../../scripts/apidocs/processing/method';
import type { RawApiDocsSignature } from '../../../scripts/apidocs/processing/signature';

function newTestMethod(
  signature: Partial<RawApiDocsSignature>
): RawApiDocsMethod {
  return {
    name: 'test',
    signatures: [
      {
        deprecated: 'deprecated',
        description: 'description',
        since: 'since',
        parameters: [],
        returns: {
          type: 'simple',
          text: 'returns',
        },
        throws: [],
        signature: 'signature',
        examples: [],
        seeAlsos: [],
        ...signature,
      },
    ],
    source: {
      filePath: 'test/page.spec.ts',
      line: 1,
      column: 1,
    },
  };
}

describe('toRefreshFunction', () => {
  it("should return 'undefined' when there are no faker calls", async () => {
    // given
    const method = newTestMethod({
      examples: ['const a = 1;'],
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toBe('undefined');
  });

  it('should handle single line calls with semicolon', async () => {
    // given
    const method = newTestMethod({
      examples: ['faker.number.int(); // 834135'],
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should handle single line calls without semicolon', async () => {
    // given
    const method = newTestMethod({
      examples: ['faker.number.int() // 834135'],
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should handle multiple calls', async () => {
    // given
    const method = newTestMethod({
      examples: ['faker.number.int()', 'faker.number.int()'],
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should handle multiline calls', async () => {
    // given
    const method = newTestMethod({
      examples: 'faker.number.int({\n  min: 1,\n  max: 10\n})'.split('\n'),
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toMatchSnapshot();
  });

  it('should handle properties after calls', async () => {
    // given
    const method = newTestMethod({
      examples: ['faker.airline.airport().name'],
    });

    // when
    const result = await toRefreshFunction(method);

    // then
    expect(result).toMatchSnapshot();
  });
});
