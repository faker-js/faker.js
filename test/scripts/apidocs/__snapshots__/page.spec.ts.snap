// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toRefreshFunction > should handle multiline calls 1`] = `
"async (): Promise<unknown[]> => {
  await enableFaker();
  faker.seed();
  faker.setDefaultRefDate();
  const result: unknown[] = [];

  try {
    result.push(
      faker.number.int({
        min: 1,
        max: 10,
      })
    );
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  return result;
}"
`;

exports[`toRefreshFunction > should handle multiple calls 1`] = `
"async (): Promise<unknown[]> => {
  await enableFaker();
  faker.seed();
  faker.setDefaultRefDate();
  const result: unknown[] = [];

  try {
    result.push(faker.number.int());
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  try {
    result.push(faker.number.int());
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  return result;
}"
`;

exports[`toRefreshFunction > should handle properties after calls 1`] = `
"async (): Promise<unknown[]> => {
  await enableFaker();
  faker.seed();
  faker.setDefaultRefDate();
  const result: unknown[] = [];

  try {
    result.push(faker.airline.airport().name);
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  return result;
}"
`;

exports[`toRefreshFunction > should handle single line calls with semicolon 1`] = `
"async (): Promise<unknown[]> => {
  await enableFaker();
  faker.seed();
  faker.setDefaultRefDate();
  const result: unknown[] = [];

  try {
    result.push(faker.number.int());
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  return result;
}"
`;

exports[`toRefreshFunction > should handle single line calls without semicolon 1`] = `
"async (): Promise<unknown[]> => {
  await enableFaker();
  faker.seed();
  faker.setDefaultRefDate();
  const result: unknown[] = [];

  try {
    result.push(faker.number.int());
  } catch (error: unknown) {
    result.push(error instanceof Error ? error.name : 'Error');
  }

  return result;
}"
`;
